components/Chips/tests/unit/ChipsTests.m

```diff
@@ -394,4 +411,40 @@ - (void)testChipsWithDeleteEnabled {
   XCTAssertEqual(controlViewCount, (NSUInteger)1);
 }
 
+- (void)testChipViewDynamicTypeBehavior {
+  // Given
+  MDCChipsTestsFakeChipView *chipView = [[MDCChipsTestsFakeChipView alloc] init];
+  UIFont *titleFont = [UIFont systemFontOfSize:14.0 weight:UIFontWeightMedium];
+  MDCFontScaler *fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
+  titleFont = [fontScaler scaledFontWithFont:titleFont];
+  titleFont = [titleFont mdc_scaledFontAtDefaultSize];
+  chipView.titleFont = titleFont;
+  chipView.mdc_adjustsFontForContentSizeCategory = YES;
+  chipView.mdc_legacyFontScaling = NO;
+  chipView.titleLabel.text = @"Chip";
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+  CGFloat defaultFontSize = chipView.titleLabel.font.pointSize;
+
+  // When
+  if (@available(iOS 10.0, *)) {
+    UIContentSizeCategory size = UIContentSizeCategoryExtraExtraLarge;
+    UITraitCollection *traitCollection =
+        [UITraitCollection traitCollectionWithPreferredContentSizeCategory:size];
+    chipView.traitCollectionOverride = traitCollection;
+  }
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+  CGFloat fontSizeForExtraExtraLargeSizeCategory = chipView.titleLabel.font.pointSize;
+
+  // Then
+  if (@available(iOS 10.0, *)) {
+    XCTAssertGreaterThan(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
+  } else {
+    XCTAssertEqual(fontSizeForExtraExtraLargeSizeCategory, defaultFontSize);
```

Comment:
Are you testing that this doesn't work on iOS 9? I think we can drop this else clause.

Reply:
It was added back from a code review comment (https://github.com/material-components/material-components-ios/pull/7394#discussion_r285156046). I felt it is okay to use it to indicate the testing behavior?

Reply:
I saw the comment below. I am going to make it only iOS10+ then.

Reply:
nvm. I saw the comment below. I am going to move all tests into iOS10+ guard.

Reply:
Addressed in 865300029f6556aaf570620546e5bcb94e64d6f6.